Add modal for selecting duration of overrides for Loop

codebymini opened the pull request:
# Overview
This PR enhances the Loop APNS override functionality by replacing the simple confirmation alert with an interactive modal that allows users to configure override duration settings before activation.
## Key Changes
### 🆕 New Features
* Override Duration Modal: Replaces basic confirmation dialog with a comprehensive modal for override configuration
* Duration Control: Added slider for setting custom override duration (15 minutes to 24+ hours)
* Indefinite Override Toggle: Added toggle to enable/disable indefinite overrides
* Smart Duration Display: Duration slider shows time in intuitive hours and minutes format (e.g., "1h 30m")
### 🔧 Enhanced User Experience
* Preset Information Display: Modal shows override details including symbol, name, target range, and insulin needs
* Contextual Warnings: Displays warning for overrides with default durations that can't be set to indefinite
* Accessible Controls: Duration controls positioned at bottom of modal for easier thumb reach
* Visual Feedback: Duration slider is disabled for overrides with preset durations
### 🏗️ Technical Improvements
* Modal Architecture: Implemented OverrideActivationModal view with proper state management
* Duration Handling: Enhanced activateOverride method to accept custom duration parameter
* State Management: Added showOverrideModal property to view model
### User Workflow Changes
Before: User taps override → Simple confirmation dialog → Override activates with preset duration

After: User taps override → Modal opens → Configure duration/indefinite setting → Override activates with chosen settings

### Benefits
* Flexibility: Users can now customize override duration instead of being limited to preset values
* Clarity: Clear visual representation of what duration will be applied
* Accessibility: Better positioned controls for mobile device usage
* Consistency: Duration format matches other parts of the app (hours and minutes)
### Files Modified
LoopFollow/Remote/LoopAPNS/OverridePresetsView.swift - Main implementation with modal and enhanced functionality
This enhancement significantly improves the user experience for remote override management while maintaining backward compatibility with existing override presets.

marionbarker commented:
## Summary

* If LoopFollow is in dark mode, I think the override modal should be in dark mode too.
* The duration slider is not working for me for override that has a set duration
* There should be a decimal point displayed for mmol/L glucose ranges

* LoopFollow phone: SE 3rd gen, iOS 18.6.1
   * Note - repeated the test on an iPhone 8 running iOS 16 and the modal was dark on that device
   * However, the duration for a defined duration override still could not be adjusted


## Test

This test was performed using a Loop URL from a phone that uses mmol/L.

**build dev branch to remind me of how the override interface works**
* when selecting the override, it asks if you want to enable it

in other words: `Before: User taps override → Simple confirmation dialog → Override activates with preset duration`

**build this branch**

**test with an indefinite override from LoopFollow override screen**
* user can just accept as it
* user can modify an indefinite override to specific duration from 15 minutes to 24 hours and return to indefinite

**test with a defined-duration override from LoopFollow override screen**
* user can just accept as it
* user should be able to modify the duration - but the slider is not working

The white full-screen modal is really high-contrast in the evening with a LoopFollow phone in dark mode.

Also, the target should show one value after the decimal point for mmol/L. The sedentary override on the Looping phone is 5,5 to 6,1.

<img width="1580" height="1374" alt="lf-pr457-formating" src="https://github.com/user-attachments/assets/be383ebf-3a74-4373-b79d-20f99f3e5086" />

marionbarker commented:
I updated the SE 3rd gen phone to iOS 18.6.2.
Either the update or the reboot associated with the update fixed the white modal problem.

codebymini commented:
> The duration slider is not working for me for override that has a set duration
This is expected behaviour. I changed the warning text and also set the Duration and Enable Indefinitly to the secondary (disabled) color
> There should be a decimal point displayed for mmol/L glucose ranges
Added formatting based on users selection of mmol/L / mg/dL

codebymini commented:
<img width="1170" height="2532" alt="IMG_7976" src="https://github.com/user-attachments/assets/c2dc46aa-f7f8-4f57-9bc4-10b268a3e016" />
<img width="1170" height="2532" alt="IMG_7977" src="https://github.com/user-attachments/assets/9a49d7f3-a9d9-45b8-a552-13f000dcbcf6" />
<img width="1170" height="2532" alt="IMG_7978" src="https://github.com/user-attachments/assets/f52182aa-45ec-4b18-b4a8-b91a52b4390e" />
* For overrides with fixed duration the toggle and slider is hidden and information about the duration is shown together with other override information.
* For overrides without duration a toggle to enable indefinitely is shown and if toggle off a slider to choose duration is shown
